### ISSUE TITLE
CopyrightBear

### ISSUE THREAD

**Issue by jayvdb** [coala/coala-bears#823]
A lint that is often missed is including copyright headers, and updating copyright years.

https://pypi.python.org/pypi/flake8-copyright

However some languages there are empty files, like python package `__init__.py` files, and no copyright header is needed.

https://github.com/savoirfairelinux/flake8-copyright/issues/1#issuecomment-138394295

**Comment by sils1297** [comment]
This could be quite simple in a language independent manner without even caring if it's a comment or not as it should be quite impossible to write code that looks like a copyright header without knowing about it.

Neat idea!

**Comment by jayvdb** [comment]
The required header can be a literal multiline value that must appear.

Different languages have different sections in .coafile, so that variation is already catered for.

Typically the copyright header varies with different names per file. Worst case dedicated sections in the .coafile support this.

Allowing patterns and variables in the template would help.

**Comment by sils1297** [comment]
@jayvdb would you be interested in doing this? I'll be available for help of course alongside <http://coala.io/writingbears>. Otherwise if it's a priority for you or someone else I'm sure we'll find someone who can look into it as it should be rather quick and rewarding.

**Comment by jayvdb** [comment]
Not a priority for me.

**Comment by jayvdb** [comment]
A script that is commonly used is `licensecheck.pl` from KDE, which made its way [into Chromium](https://github.com/adobe/chromium/blob/master/third_party/devscripts/licensecheck.pl), and I've noticed it just now appearing in https://github.com/html5lib/html5lib-python/pull/303

**Comment by sils** [comment]
@jayvdb do you think it makes sense to try to be better than licensecheck.pl or just wrap it? We could easily include it in the distribution I guess.

**Comment by jayvdb** [comment]
@sils, looks like we can wrap https://packages.debian.org/sid/licensecheck . With Debian and Chromium using it, I think using the same maintained script is a good approach that helps packagers now.
If someone wants to rewrite it, or build a feature complete equivalent, yay.

**Comment by Adrianzatreanu** [comment]
hey @divyam3897 still on this :D ?

**Comment by divyam3897** [comment]
@Adrianzatreanu yes, stuck on few things which I am unable to figure out :( . I should commit what I am trying ? Though I have been trying to understand the workflow of the bears by going throught the bears and docs. Sorry, its taking too long :(

**Comment by Makman2** [comment]
Just visit us on gitter, there we can help you :)

**Comment by yash-nisar** [comment]
Still on it @divyam3897 ?

**Comment by divyam3897** [comment]
@yash-nisar didn't get the time to hack on it. Feel free to assign it to someone else to make it done faster. I can review the commits, progress and help with my so far progress :)

**Comment by yash-nisar** [comment]
I would like to work on this @Makman2 @sils

**Comment by yash-nisar** [comment]
IIUC, the bear will check for copyright notices in all python files by running a simple regex search for strings like :
`Copyright 2014 <author>`
`Copyright (C) 2014 <author>`
CC @Makman2 @sils @jayvdb

**Comment by Makman2** [comment]
imo this is up to you ;)

**Comment by yash-nisar** [comment]
Can we rename is to LicenseCheckBear and create a new bear for checking Copyright ?

**Comment by Makman2** [comment]
If the issue is not completely solved, let's reopen it :)

**Comment by yash-nisar** [comment]
So the milestone now would be to add CopyrightBear ?

**Comment by jayvdb** [comment]
The existing external linter does both. The bear just needs improving.

**Comment by yash-nisar** [comment]
We won't be able to create user defined regexes IIUC and will have to rely on the regex defined in the script. Is that okay ?

**Comment by jayvdb** [comment]
Many very important codebases use this external tool, so we need to expose its functionality via coala as sensibly as possible.

If someone wants a different regex, they can write a RegexBear, or their own copyright bear. ;)

Or they can enhance the external tool and then improve our bear.

**Comment by yash-nisar** [comment]
So, we should match the regex `*No copyright*` and display the output message `No copyright found` or similar if the user sets this `check_copyright` option as `True`. Is that what you imply @jayvdb ?

**Comment by Makman2** [comment]
I don't imply anything, but sounds reasonable 👍  :)

**Comment by yash-nisar** [comment]
Well, I will have to override `process_output` in this case too and match regexes for license as well as copyright notice. :)

**Comment by khanchi97** [comment]
Hello @jayvdb I want to write this bear but I'm not confirmed which linter I have to wrap.The one that you gave https://pypi.python.org/pypi/flake8-copyright is only for python files but this bear would be a `genericbear`.So can you tell me which linter I have to use?
